### PR TITLE
Ensure buffered string to have length 32

### PR DESCRIPTION
**Pull request by HoOngEe**
`bitcore-mnemonic` library overrides `toBuffer` action so that the starting null bytes are omitted. So I should ensure the strings have length 32 before they are hashed.

### CODE REVIEW

File: src/key/schnorr.ts [hunk 1]
```diff
@@ -25,9 +25,16 @@ export interface SchnorrSignature {
...
+export function schnorrHash(r: BN, msg: BN): string {
+    const rString = r
+        .toBuffer("be", 32)
+        .toString("binary")
+        .padStart(32, "\x00");
+    const mString = msg
+        .toBuffer("be", 32)
+        .toString("binary")
+        .padStart(32, "\x00");
+    console.log(msg.toBuffer("be", 32).length);
```
Comment (majecty) [inline src/key/schnorr.ts]: Please remove this line.

File: src/key/schnorr.ts [hunk 2]
```diff
@@ -25,9 +25,16 @@ export interface SchnorrSignature {
...
-    const mString = msg.toBuffer("be", 32).toString("binary");
+export function schnorrHash(r: BN, msg: BN): string {
+    const rString = r
+        .toBuffer("be", 32)
+        .toString("binary")
+        .padStart(32, "\x00");
+    const mString = msg
+        .toBuffer("be", 32)
+        .toString("binary")
+        .padStart(32, "\x00");
```
Comment (majecty) [inline src/key/schnorr.ts]: Please add a simple comment about why we need padding.

File: src/key/schnorr.ts (hunk 1 above)
Comment (HoOngEe) [reply src/key/schnorr.ts]: Fixed

File: src/key/schnorr.ts (hunk 2 above)
Comment (HoOngEe) [reply src/key/schnorr.ts]: I added some explanation

**Pull request merged**